app/assets/javascripts/Components/Result/notebook_viewer.jsx

```diff
@@ -2,27 +2,39 @@ import React from "react";
 import {markupTextInRange} from "../Helpers/range_selector";
 
 export class NotebookViewer extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      annotations: [],
-    };
+  componentDidMount() {
+    if (this.props.resultView) {
+      this.readyAnnotations();
+    }
   }
 
-  getSelection = () => {
+  readyAnnotations = () => {
+    annotation_type = ANNOTATION_TYPES.NOTEBOOK;
+  };
+
+  renderAnnotations = () => {
     const iframe = document.getElementById("notebook");
-    const target = iframe.contentWindow || iframe.contentDocument;
-    const range = target.getSelection().getRangeAt(0);
-    this.setState(prevState => ({
-      annotations: prevState.annotations.concat([range]),
-    }));
-    markupTextInRange(range, "yellow");
+    const doc = iframe.contentDocument;
+    this.props.annotations.forEach(annotation => {
+      const start_node = doc.evaluate(annotation.start_node, doc).iterateNext();
+      const end_node = doc.evaluate(annotation.end_node, doc).iterateNext();
+      const newRange = doc.createRange();
+      newRange.setStart(start_node, annotation.start_offset);
+      newRange.setEnd(end_node, annotation.end_offset);
+      markupTextInRange(newRange, "yellow", annotation.content);
+    });
   };
 
   render() {
     return (
       <div>
-        <iframe className={"notebook"} id={"notebook"} src={this.props.url + "&preview=true"} />
+        <iframe
+          className={"notebook"}
+          id={"notebook"}
+          key={this.props.annotations} // reload the iframe when the annotations change
```

Comment:
I'm not sure exactly what's going on, but when you edit an annotation in the table, the annotation text doesn't change in the notebook viewer. (Unless you reload the page, of course.)

But if you delete an annotation from the table, it does get removed when switching back to the notebook viewer.